lib/ip.go

```diff
@@ -37,34 +50,93 @@ func RandIP4List(n int) []net.IP {
 	return ips
 }
 
-// RandIP4Range returns a list of randomly generated IPv4 addresses within
-// starting and ending IPrange.
+// EvalIP4Range checks if the starting and ending IP range is valid or not.
+//
+// note: starting and ending IPs must be valid IPv4 string formats.
+func EvalIP4Range(
+	startIP string,
+	endIP string,
+) (IP4Range, error) {
+	startIPRaw := net.ParseIP(startIP).To4()
+	if len(startIPRaw) == 0 || len(startIPRaw) > net.IPv4len {
+		return IP4Range{}, errors.New("invalid range start IP")
+	}
 
-// note: starting and ending IPs should be in format of "1.1.1.1"
-func RandIP4Range(startIP, EndIP string) (net.IP, error) {
-	StartIPRaw := net.ParseIP(startIP).To4()
-	if len(StartIPRaw) == 0 || len(StartIPRaw) > net.IPv4len {
-		return nil, fmt.Errorf("range is Invalid")
+	startIPInt := binary.BigEndian.Uint32(startIPRaw.To4())
```

Comment:
This is only used later in the code, after the error below this occurs. So you can initialize it right before `endIPInt`.

Reply:
Additionally, `startIPRaw` was already generated by calling `.To4()`, so this is redundant. Same for `endIPRaw` below when generating `endIPInt`.